src/client/http.rs

```diff
@@ -40,36 +40,36 @@ impl<'a> HttpClient<'a> {
         }
     }
 
-    pub fn add_host(&mut self, host: &'a str) {
-        self.hosts.push(host);
+    pub fn add_host<S>(&mut self, host: S) where S: Into<String> {
+        self.hosts.push(host.into());
     }
 
-    fn get_host(&self) -> &'a str {
+    fn get_host(&self) -> String {
         match self.hosts.first() {
-            Some(host) => host,
+            Some(host) => host.to_string(),
             None => panic!("Could not get host")
         }
     }
 }
 
-impl<'a> Client for HttpClient<'a> {
+impl Client for HttpClient {
     fn query(&self, q: String, epoch: Option<Precision>) -> ClientReadResult {
         let host = self.get_host();
 
         let mut query = HashMap::new();
-        query.insert("db", self.credentials.database.to_string());
-        query.insert("q", q);
+        query.insert(String::from("db"), self.credentials.database.to_string());
+        query.insert(String::from("q"), q);
 
         if let Some(ref epoch) = epoch {
-            query.insert("epoch", epoch.to_string());
+            query.insert(String::from("epoch"), epoch.to_string());
         }
 
         let request = Request {
-            url: &*{host.to_string() + "/query"},
+            url: format!("{}/query", host.to_string()),
             method: Method::GET,
             auth: Some(Auth {
-                username: self.credentials.username,
-                password: self.credentials.password
+                username: self.credentials.username.clone(),
```

Comment:
Cloning here is not pretty. How can this be improved?

Reply:
Maybe `Cow<&str>`?